chat/internal/app/repositories/chat_repo/impl.go

```diff
@@ -0,0 +1,300 @@
+package chat_repo
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/BurMachine/Bigtech_microservices/chat/internal/app/models"
+	"github.com/BurMachine/Bigtech_microservices/chat/pkg/postgres"
+	"github.com/Masterminds/squirrel"
+	"github.com/google/uuid"
+)
+
+// CreateDirectChat создает новый чат и добавляет участников
+func (r *Repository) CreateDirectChat(ctx context.Context, chat *models.Chat) (string, error) {
+	const api = "chat_repo.Repository.CreateDirectChat"
+
+	// Проверка входных данных
+	if len(chat.Participants) < 2 {
+		return "", fmt.Errorf("%s: %w", api, ErrRepoInvalidArg)
+	}
+
+	// Проверка валидности UUID участников
+	for _, userID := range chat.Participants {
+		if _, err := uuid.Parse(userID); err != nil {
+			return "", fmt.Errorf("%s: invalid user_id format: %w", api, ErrRepoInvalidArg)
+		}
+	}
+
+	// Генерация UUID для чата
+	chatID := uuid.New().String()
+
+	// Создание записи чата
+	qb := r.qb.Insert(tableChats).
+		Columns(colChatID, colChatCreatedAt).
+		Values(chatID, time.Now())
+
+	conn := r.db.GetQueryEngine(ctx)
+	if _, err := conn.Execx(ctx, qb); err != nil {
+		return "", fmt.Errorf("%s: %w", api, postgres.ConvertPGError(err))
+	}
+
+	// Добавление участников
+	for _, userID := range chat.Participants {
+		qb = r.qb.Insert(tableChatMembers).
+			Columns(colChatMemberChatID, colChatMemberUserID).
+			Values(chatID, userID)
+		if _, err := conn.Execx(ctx, qb); err != nil {
+			if IsUniqueViolation(err) {
+				return "", fmt.Errorf("%s: %w", api, errRepoAlreadyExists)
+			}
+			return "", fmt.Errorf("%s: %w", api, postgres.ConvertPGError(err))
```

Comment:
Вставка в БД в цикле - это плохо. Можно сделать одним запросом:



```suggestion
    qb = r.qb.Insert(tableChatMembers).
			Columns(colChatMemberChatID, colChatMemberUserID)

	// Добавление участников
	for _, userID := range chat.Participants {
		qb = qb.Values(chatID, userID)
	}
    if _, err := conn.Execx(ctx, qb); err != nil {
     ///
    }
```